src/test/java/com/google/jenkins/plugins/storage/IT/ITUtil.java

```diff
@@ -0,0 +1,156 @@
+   */
+  static void dumpLog(Logger logger, Run<?, ?> run) throws IOException {
+    BufferedReader reader = new BufferedReader(run.getLogReader());
+    String line = null;
+    while ((line = reader.readLine()) != null) {
+      logger.info(line);
+    }
+  }
+
+  /**
+   * Retrieves the location set through environment variables
+   *
+   * @return A Google Compute Resource Region (us-west1) or Zone (us-west1-a) string
+   */
+  static String getLocation() {
+    String location = System.getenv("GOOGLE_PROJECT_LOCATION");
+    if (location == null) {
+      location = System.getenv("GOOGLE_PROJECT_ZONE");
+    }
+    assertNotNull("GOOGLE_PROJECT_LOCATION env var must be set", location);
+    return location;
+  }
+
+  /**
+   * Given jenkins instance and credentials ID, return Credential
+   *
+   * @param itemGroup A handle to the Jenkins instance.
+   * @param credentialsId credentialsId to retrieve credential. Must exist in credentials store.
+   * @return Credential based on credentialsId.
+   * @throws GeneralSecurityException
+   */
+  static Credential getCredential(ItemGroup itemGroup, String credentialsId)
+      throws GeneralSecurityException {
+    GoogleRobotCredentials robotCreds =
+        CredentialsMatchers.firstOrNull(
+            CredentialsProvider.lookupCredentials(
+                GoogleRobotCredentials.class, itemGroup, ACL.SYSTEM, new ArrayList<>()),
+            CredentialsMatchers.withId(credentialsId));
+    try {
+      return robotCreds.getGoogleCredential(new StorageScopeRequirement());
+    } catch (GeneralSecurityException gse) {
+      throw new GeneralSecurityException(gse);
+    }
+  }
+
+  /**
+   * Given Jenkins instance and credentials ID, return a Storage object to make calls to the Google
+   * Cloud Storage JSON API.
+   *
+   * @param itemGroup A handle to the Jenkins instance.
+   * @param credentialsId credentialsId to retrieve credential. Must exist in credentials store.
+   * @return Storage object authenticated with credentialsId.
+   * @throws GeneralSecurityException
+   */
+  static Storage getService(ItemGroup itemGroup, String credentialsId)
+      throws GeneralSecurityException {
+    try {
+      return new Storage.Builder(
+              new NetHttpTransport(), new JacksonFactory(), getCredential(itemGroup, credentialsId))
+          .build();
+    } catch (GeneralSecurityException gse) {
+      throw new GeneralSecurityException(gse);
+    }
+  }
+
+  /**
+   * Delete object matching pattern from Google Cloud Storage bucket of name bucket.
+   *
+   * @param itemGroup A handle to the Jenkins instance.
+   * @param credentialsId credentialsId to retrieve credential. Must exist in credentials store.
+   * @param bucket Name of Google Cloud Storage bucket to delete from.
+   * @param pattern Pattern to match object name to delete from bucket.
+   * @throws GeneralSecurityException
+   * @throws IOException
+   */
+  static void deleteFromBucket(
```

Comment:
This and other related methods should be abstracted out into a StorageClient class.

Reply:
Done.